core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1176,6 +1198,19 @@ public Builder removeBranch(String branch) {
       return this;
     }
 
+    public Builder setStatistics(String snapshotId, @Nullable StatisticsFile statisticsFile) {
+      Preconditions.checkNotNull(snapshotId, "snapshotId is null");
+      Preconditions.checkNotNull(statisticsFile, "statisticsFile is null");
+      this.statistics.put(snapshotId, statisticsFile);
+      return this;
```

Comment:
If these methods don't add changes to the `changes` list, then the builder won't recognize that anything has been updated and will return the original metadata. This should be picked up by tests of the table metadata updates.

I think you should probably move the changes to this file (and change additions) to a separate PR. We're going to need to add `MetadataUpdate` types for this, and update the JSON parser for serializing those updates. That, and testing everything. This is probably best separated from the API changes, which can be done independently without an implementation.

Reply:
moved to https://github.com/apache/iceberg/pull/5450

Reply:
Can you perhaps point me to existing test coverage I should follow for these?

Reply:
@findepi, the newer tests are going into `TestTableMetadata`. Before the builder, `TableMetadata` exposed methods like `replaceSortOrder` that are tested directly (that's why it's hard to find the tests). And quite a few things are still tested end-to-end using the table operations themselves (like `UpdateSchema`). I'd take a look at the tests for `replaceSortOrder` in `TestTableMetadata`.

Reply:
Thanks @rdblue for the pointer.
FYI, per the suggestion, i backed out `TableMetadata` changes from this PR. Will address these tests in the other one.
Can I assume the tests for `TableMetadata` changes are not prerequisite for this PR being merged?